src/test/components/ThreadActivityGraph.test.js

```diff
@@ -379,6 +379,47 @@ describe('ThreadActivityGraph', function () {
       )
     ).toEqual([]);
   });
+
+  it('selects the correct call node path when clicked on an area where multiple stacks overlap with various categories', function () {
```

Comment:
It took me a while to come up with a test case that fails on the previous code but I managed to create one. It fails now if we revert the fix.